blockchain/indexbuilder.go

```diff
@@ -113,12 +119,144 @@ func (ib *IndexBuilder) HandleBlock(blk *block.Block) error {
 	ib.pendingBlks <- blk
 	return nil
 }
-
+func initIndexActionsKey(store db.KVStore) error {
+	_, err := store.Get(blockActionBlockMappingNS, indexActionsKey)
+	if err != nil && errors.Cause(err) == db.ErrNotExist {
+		if err = store.Put(blockActionBlockMappingNS, indexActionsKey, make([]byte, 8)); err != nil {
+			return errors.Wrap(err, "failed to write initial value for index actions")
+		}
+		return nil
+	}
+	return err
+}
+func (ib *IndexBuilder) getStartHeightAndIndex(tipHeight uint64) (startHeight, startIndex uint64, err error) {
+	// get index that already builded
+	startIndex, err = getNextIndex(ib.store)
+	if err != nil {
+		return
+	}
+	currentNumOfActions := uint64(0)
+	startHeight = uint64(1)
+	for i := uint64(1); i <= tipHeight; i++ {
```

Comment:
It's heavy to loop through to find what block is the last indexed one.

Instead, you could add a k/v pair to remember it, and update this value every time we have successfully indexed one block

Reply:
sure,I'll add